shared/test/test_assets.rb

```diff
@@ -56,7 +56,17 @@ def test_assets
     delete(@assets, channel_id, 'filename.jpg')
     assert @assets.last_response.successful?
 
-    # invalid files are not uploaded
+    # file extension case insensitivity
+    put(@assets, channel_id, 'filename.JPG', 'stub-contents', 'application/jpeg')
+    assert @assets.last_response.successful?
+    get(@assets, channel_id, 'filename.JPG')
+    assert @assets.last_response.successful?
+    get(@assets, channel_id, 'filename.jpg')
+    assert @assets.last_response.not_found?
```

Comment:
Either seems ok to me since I think it will be relatively uncommon for people to upload images which differ only in the case of the extension. Admittedly I chose the quickest option. If we want to downcase the extension, RESTful ways to do it include:
1. downcase the extension on the client before sending the PUT request
2. turn this into a POST request containing the requested filename, and return the new/actual filename (like when creating a channel)

(2) seems heavy-handed unless we know we're going to want to move to this kind of model in the future (e.g. if we are adding content-based hashes to the filenames), but if (1) downcasing the extension on the client sounds better to you I'd be happy to do it that way instead.